assembly/src/parsers/mod.rs

```diff
@@ -436,6 +453,15 @@ fn parse_constant(token: &Token) -> Result<(String, u64), ParsingError> {
     }
 }
 
+// PARSED BODY
+// ================================================================================================
+
+#[derive(Clone, Default, PartialEq, Eq, Debug)]
+pub struct ParsedCode {
+    pub nodes: Vec<Node>,
+    pub locations: Vec<SourceLocation>,
+}
```

Comment:
nit: let's rename it to `CodeBody` after all - I think it is the best out of the alternatives - though, I still don't love it.